demisto_sdk/commands/format/format_module.py

```diff
@@ -424,36 +426,71 @@ def format_output(
-            skipped_list.append(f"Validate Status on file: {input} - Skipped")
-        elif validate_res == VALIDATE_RES_FAILED_CODE:
-            error_list.append(f"Format Status on file: {input} - Failed")
-        else:
-            error_list.append(f"Format Status on file: {input} - Failed")
-            error_list.append(f"Validate Status on file: {input} - Failed")
-    elif format_res and not validate_res:
-        error_list.append(f"Format Status on file: {input} - Failed")
-        info_list.append(f"Validate Status on file: {input} - Success")
-    elif not format_res and validate_res:
-        if validate_res == VALIDATE_RES_SKIPPED_CODE:
-            info_list.append(f"Format Status on file: {input} - Success")
-            skipped_list.append(f"Validate Status on file: {input} - Skipped")
-        elif validate_res == VALIDATE_RES_FAILED_CODE:
-            info_list.append(f"Format Status on file: {input} - Success")
-        else:
-            info_list.append(f"Format Status on file: {input} - Success")
-            error_list.append(f"Validate Status on file: {input} - Failed")
+
+    def _create_status_msg(operation: str, status: str) -> str:
+        return f"{operation} Status on file: {input} - {status}"
+
+    # Determine format status
+    if format_res == SUCCESS_RETURN_CODE:
+        format_status = "success"
+        format_msg = _create_status_msg("Format", "Success")
+    elif format_res == SKIP_RETURN_CODE:
+        format_status = "skipped"
+        format_msg = _create_status_msg("Format", "Skipped")
+    else:
+        format_status = "failed"
+        format_msg = _create_status_msg("Format", "Failed")
+
+    # Determine validation status
+    if validate_res == SUCCESS_RETURN_CODE:
+        validate_status = "success"
+        validate_msg = _create_status_msg("Validate", "Success")
+    elif validate_res == SKIP_RETURN_CODE:
+        validate_status = "skipped"
+        validate_msg = _create_status_msg("Validate", "Skipped")
+    elif validate_res == VALIDATE_RES_FAILED_CODE:
+        validate_status = "failed_no_validation"
+        validate_msg = None  # No validation message when validation wasn't run
+    else:
+        validate_status = "failed"
+        validate_msg = _create_status_msg("Validate", "Failed")
+
+    # Add format message to appropriate list
+    if format_status == "success":
+        info_list.append(format_msg)
+    elif format_status == "skipped":
+        skipped_list.append(format_msg)
+    else:  # failed
+        error_list.append(format_msg)
+
+    # Add validation message to appropriate list
+    if validate_status == "success":
+        info_list.append(validate_msg)
+    elif validate_status == "skipped":
+        skipped_list.append(validate_msg)
+    elif validate_status == "failed":
+        error_list.append(validate_msg)
+        # Only add help message if format succeeded and validation failed
+        if format_status == "success":
```

Comment:
The splitting into 2 parts (for format and for validate) is good, but we can still create and save the messages to the lists in the same place, instead of saving the value to a variable. The function you added is what simplifies it even more.